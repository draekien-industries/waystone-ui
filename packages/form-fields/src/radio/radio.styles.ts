import { cssSelectors } from '@waystone/css-presets';
import type { ThemeUICSSObject } from 'theme-ui';

export const containerCss = {
  gap: 'small',
} as const satisfies ThemeUICSSObject;

export const inputCss = {
  height: '1.25rem',
  width: '1.25rem',
  cursor: 'pointer',
  [cssSelectors.disabled]: {
    cursor: 'not-allowed',
  },
} as const satisfies ThemeUICSSObject;

export const labelCss = {
  cursor: 'pointer',
  [cssSelectors.disabled]: {
    cursor: 'not-allowed',
  },
} as const satisfies ThemeUICSSObject;
