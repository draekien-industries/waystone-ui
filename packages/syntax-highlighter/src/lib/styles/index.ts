export * from './a11yDark.generated';
export * from './a11yLight.generated';
export * from './agate.generated';
export * from './anOldHope.generated';
export * from './androidstudio.generated';
export * from './arduinoLight.generated';
export * from './arta.generated';
export * from './ascetic.generated';
export * from './atomOneDarkReasonable.generated';
export * from './atomOneDark.generated';
export * from './atomOneLight.generated';
export * from './brownPaper.generated';
export * from './codepenEmbed.generated';
export * from './colorBrewer.generated';
export * from './dark.generated';
export * from './defaultStyle.generated';
export * from './devibeans.generated';
export * from './docco.generated';
export * from './far.generated';
export * from './felipec.generated';
export * from './foundation.generated';
export * from './githubDarkDimmed.generated';
export * from './githubDark.generated';
export * from './github.generated';
export * from './gml.generated';
export * from './googlecode.generated';
export * from './gradientDark.generated';
export * from './gradientLight.generated';
export * from './grayscale.generated';
export * from './hybrid.generated';
export * from './idea.generated';
export * from './intellijLight.generated';
export * from './irBlack.generated';
export * from './isblEditorDark.generated';
export * from './isblEditorLight.generated';
export * from './kimbieDark.generated';
export * from './kimbieLight.generated';
export * from './lightfair.generated';
export * from './lioshi.generated';
export * from './magula.generated';
export * from './monoBlue.generated';
export * from './monokaiSublime.generated';
export * from './monokai.generated';
export * from './nightOwl.generated';
export * from './nnfxDark.generated';
export * from './nnfxLight.generated';
export * from './nord.generated';
export * from './obsidian.generated';
export * from './pandaSyntaxDark.generated';
export * from './pandaSyntaxLight.generated';
export * from './paraisoDark.generated';
export * from './paraisoLight.generated';
export * from './pojoaque.generated';
export * from './purebasic.generated';
export * from './qtcreatorDark.generated';
export * from './qtcreatorLight.generated';
export * from './rainbow.generated';
export * from './routeros.generated';
export * from './schoolBook.generated';
export * from './shadesOfPurple.generated';
export * from './srcery.generated';
export * from './stackoverflowDark.generated';
export * from './stackoverflowLight.generated';
export * from './sunburst.generated';
export * from './tokyoNightDark.generated';
export * from './tokyoNightLight.generated';
export * from './tomorrowNightBlue.generated';
export * from './tomorrowNightBright.generated';
export * from './vs.generated';
export * from './vs2015.generated';
export * from './xcode.generated';
export * from './xt256.generated';
