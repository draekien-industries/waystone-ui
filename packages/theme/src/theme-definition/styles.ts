export const styles = {
  shadows: {
    none: 'none',
    sm: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    md: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    lg: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
} as const;

export default styles;
