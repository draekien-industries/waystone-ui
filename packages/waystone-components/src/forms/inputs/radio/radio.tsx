import type { HTMLProps } from 'react';
import { forwardRef } from 'react';

export type RadioProps = {
  name: string;
} & Omit<HTMLProps<HTMLInputElement>, 'type' | 'name'>;

export const Radio = forwardRef<HTMLInputElement, RadioProps>((props, ref) => (
  <input ref={ref} type="radio" {...props} />
));

Radio.displayName = 'Radio';

export default Radio;
