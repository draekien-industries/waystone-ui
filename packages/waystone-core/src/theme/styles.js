export const styles = {
    shadows: {
        sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
        md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
        lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
        'sm-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.125em 0.25em -0.125em #0b0b0bA0',
        'md-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.5em 1em -0.5em #0b0b0bA0',
        'lg-dark': '0 0 0.5em 0 #1b1b1bFF, 0 1em 2em -1em #0b0b0bA0',
    },
    sizes: {
        xs: '1em',
        sm: '5em',
        md: '10em',
        lg: '15em',
        xl: '20em',
        half: '50%',
        full: '100%',
    },
};
export default styles;
