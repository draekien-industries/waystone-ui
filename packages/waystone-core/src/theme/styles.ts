export const styles = {
  shadows: {
    sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
    md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
    lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
    'sm-dark':
      '0 0 0.5em 0 rgba(27, 27, 27, 1), 0 0.5 1em -0.5em rgba(11, 11, 11, 0.73)',
    'md-dark':
      '0 0 1em 0 rgba(27, 27, 27, 1), 0 1em 2em -1em rgba(11, 11, 11, 0.73)',
    'lg-dark':
      '0 0 2em 0 rgba(27, 27, 27, 1), 0 2em 4em -2em rgba(11, 11, 11, 0.73)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
};

export default styles;
