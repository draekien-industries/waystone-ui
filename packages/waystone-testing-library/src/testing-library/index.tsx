'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
