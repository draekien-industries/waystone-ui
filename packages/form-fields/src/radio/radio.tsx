import type { FullWidthAttributes } from '@waystone/types';
import { type ForwardedRef, forwardRef } from 'react';
import { Flex } from 'theme-ui';
import type {
  InputAriaAttributes,
  InputCallbackAttributes,
  SharedInputAttributes,
} from '../types';
import { Label } from '../label';
import { containerCss, inputCss, labelCss } from './radio.styles';

export type RadioProps = {
  label?: string;
  checked?: boolean;
  defaultChecked?: boolean;
} & InputAriaAttributes &
  InputCallbackAttributes &
  SharedInputAttributes &
  FullWidthAttributes;

const RadioWithRef = (
  { id, label, ...rest }: RadioProps,
  ref: ForwardedRef<HTMLInputElement>
) => (
  <Flex sx={containerCss}>
    <input ref={ref} sx={inputCss} type="radio" {...rest} />
    {label && (
      <Label sx={labelCss} htmlFor={id}>
        {label}
      </Label>
    )}
  </Flex>
);

export const Radio = forwardRef(RadioWithRef);
