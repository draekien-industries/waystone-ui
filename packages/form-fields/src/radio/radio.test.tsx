import { render } from '@waystone/testing-library';
import { Radio } from './radio';
import { containerCss, inputCss, labelCss } from './radio.styles';

describe('<Radio />', () => {
  describe('component', () => {
    it('should match snapshot', () => {
      const { container } = render(<Radio id="test-radio" label="Option 1" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('css', () => {
    describe('container', () => {
      it('should match snapshot', () => {
        expect(containerCss).toMatchSnapshot();
      });
    });

    describe('input', () => {
      it('should match snapshot', () => {
        expect(inputCss).toMatchSnapshot();
      });
    });

    describe('label', () => {
      it('should match snapshot', () => {
        expect(labelCss).toMatchSnapshot();
      });
    });
  });
});
