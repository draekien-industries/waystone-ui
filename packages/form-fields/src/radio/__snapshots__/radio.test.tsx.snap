// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> component should match snapshot 1`] = `
<div>
  <div
    class="css-100ak8h"
  >
    <input
      class="css-17bcjra"
      type="radio"
    />
    <label
      class="css-15xkxno"
      for="test-radio"
    >
      Option 1
    </label>
  </div>
</div>
`;

exports[`<Radio /> css container should match snapshot 1`] = `
{
  "gap": "small",
}
`;

exports[`<Radio /> css input should match snapshot 1`] = `
{
  ":after": {
    "border": "4px solid",
    "borderColor": "transparent",
    "borderRadius": "50%",
    "content": """",
    "inset": -6,
    "position": "absolute",
    "transition": "border-color 200ms",
  },
  ":disabled": {
    "cursor": "not-allowed",
  },
  ":enabled:active": {
    ":after": {
      "borderColor": [Function],
    },
  },
  ":enabled:not(active):focus-visible": {
    ":after": {
      "borderColor": [Function],
    },
  },
  "cursor": "pointer",
  "height": "1.25rem",
  "outlineColor": [Function],
  "outlineOffset": 2,
  "outlineStyle": "solid",
  "outlineWidth": 4,
  "position": "relative",
  "width": "1.25rem",
}
`;

exports[`<Radio /> css label should match snapshot 1`] = `
{
  ":disabled": {
    "cursor": "not-allowed",
  },
  "cursor": "pointer",
}
`;
